packages/xo-web/src/xo-app/vm/tab-disks.js

```diff
@@ -670,6 +754,31 @@ export default class TabDisks extends Component {
       isAdmin || (resourceSet == null && isVmAdmin)
   )
 
+  _getHostSrShouldBeOn = createSelector(
+    () => this.props.srs,
+    () => this.props.vdis,
+    (srs, vdis) => {
+      let container
+      let noPermissionSr = false
+      let sharedSr = 0
+      let sr
+      return every(vdis, ({ $SR: srId }) => {
+        sr = srs[srId]
+        return sr === undefined
+          ? ((noPermissionSr = true), false)
+          : isSrShared(sr)
+          ? (sharedSr++, true)
+          : container === undefined
+          ? ((container = sr.$container), true)
+          : sr.$container === container
+      })
```

Comment:
`every` can test the compatibility between the VDIs without doing extra tests.

Reply:
Thank you for the clarification, it's very helpful :+1:
 
I see that this method can returns a `boolean` or a `string`. To simplify it, i suggest this solution:

```js
// result:
// 'containerId': SRs are compatible (not show warning)
// '': all SRs are shared or the user has not a permission on a SR (not show warning)
// undefined: VM is broken (show warning)
_getHostSrShouldBeOn = createSelector(
  () => this.props.srs,
  () => this.props.vdis,
  (srs, vdis) => {
    let container = ''
    for (const id in vdis) {
      const sr = srs[vdis[id].$SR]
      // no permission
      if (sr === undefined) {
        container = ''
        break
      }

      if (isShared(sr)) {
        continue
      }

      // VM is broken
      if (container !== '' && container !== sr.$container) {
        container = undefined
        break
      }

      container = sr.$container
    }
    return container
  }
)
```